### PR TITLE
Code alignment with rest of the SSV code + Unit Tests

### PR DESCRIPTION
**Pull request by aayush-rockx**
Changes

- [x] Refactor Msg Store to Msg Container
- [x] Align Naming convention with rest of the SSV code

### CODE REVIEW

File: dkg/frost/instance.go [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+func NewResharing(
+	network dkg.Network,
+	operatorID types.OperatorID,
+	requestID dkg.RequestID,
+	signer types.DKGSigner,
+	storage dkg.Storage,
+	operatorsOld []types.OperatorID,
+	init *dkg.Reshare,
+	output *dkg.KeyGenOutput,
+) dkg.Protocol {
```
Comment (gl-rockx) [inline dkg/frost/instance.go]: Since we already create `Config` and `InstanceParams`, we don't need these two `New` functions anymore. We can use init/reshare to create `InstanceParams` and the `New` will just require `Config` and `InstanceParams`.

File: dkg/frost/instance.go [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+		threshold:       uint32(init.Threshold),
+		operatorID:      operatorID,
+		operators:       types.OperatorList(init.OperatorIDs).ToUint32List(),
+		operatorsOld:    types.OperatorList(operatorsOld).ToUint32List(),
+		oldKeyGenOutput: output,
+	}
+	return newProtocol(network, signer, storage, instanceParams)
+}
+
+func newProtocol(network dkg.Network, signer types.DKGSigner, storage dkg.Storage, instanceParams InstanceParams) dkg.Protocol {
```
Comment (gl-rockx) [inline dkg/frost/instance.go]: Pass config as argument directly, prepare it before calling `New`. This makes mocking etc more convenient.

File: dkg/frost/instance_params.go [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+// operatorID, threshold, operator list etc.
+type InstanceParams struct {
+	identifier      dkg.RequestID
+	threshold       uint32
+	operatorID      types.OperatorID
+	operators       []uint32
+	operatorsOld    []uint32
+	oldKeyGenOutput *dkg.KeyGenOutput
+}
+
```
Comment (gl-rockx) [inline dkg/frost/instance_params.go]: Better add some description about fields.

File: dkg/frost/round1.go [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+	}
+
+	bcastMsg, err := fr.saveSignedMsg(msg)
+	if err != nil {
+		return false, nil, err
+	}
+	return false, nil, fr.config.network.BroadcastDKGMessage(bcastMsg)
+}
+
+func (fr *Instance) partialInterpolate() ([]byte, error) {
```
Comment (gl-rockx) [inline dkg/frost/round1.go:83]: Maybe add some description, especially the formula.

File: dkg/frost/instance.go [hunk 5]
```diff
@@ -0,0 +1,244 @@
...
+func init() {
+	types.InitBLS()
+}
+
+// Instance is a single FROST DKG instance. It implements Start to init DKG
+// parameters and broadcast session key, and ProcessMsg to process 2 rounds of
+// frost, blame and keygen output messages
+type Instance struct {
+	state  *State
+	config *Config
```
Comment (gl-rockx) [inline dkg/frost/instance.go]: We need to use `IConfig` instead of `*Config`.

File: dkg/frost/messages.go [hunk 6]
```diff
@@ -0,0 +1,239 @@
...
+}
+
+type BlameType int
+
+const (
+	// InconsistentMessage refers to an operator sending multiple messages for same round
+	InconsistentMessage BlameType = iota
+	// InvalidShare refers to an operator sending invalid share
+	InvalidShare
+	//// InvalidMessage refers to messages containing invalid values
```
Comment (gl-rockx) [inline dkg/frost/messages.go]: Too many forward slashes.

File: dkg/frost/keygen_output.go [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+// verifyShares reconstructs Vk over a sliding window of t (threshold) operators
+// and checks if reconstructed Vk is consistent
+func (fr *Instance) verifyShares() (*bls.G1, error) {
+
+	var (
+		quorumStart       = 0
+		prevReconstructed = (*bls.G1)(nil)
+	)
+
+	// Sliding window of quorum 0...threshold until n-threshold...n
```
Comment (gl-rockx) [inline dkg/frost/keygen_output.go]: Add the rationale: by using a sliding window, every share is being used at least once, if every reconstruction produces an identical Vk, we can tell all the shares are valid.

**Review by gl-rockx — COMMENTED** [review]
Do we have more edge cases to test? Maybe we can do a more thorough check after we handle timeout. I think we can add more comments about the steps in the functions. This also helps us make sure all branches of logics are covered.

One other thing we need to consider is how to make sure all the secrets are wiped out when the instance is dropped. This can be a separate issue to work on.

File: dkg/frost/state.go [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+}
+
+// State tracks protocol's current round, stores messages in MsgContainer, stores
+// session key and operator's secret shares
+type State struct {
+	currentRound   ProtocolRound
+	participant    *frost.DkgParticipant
+	sessionSK      *ecies.PrivateKey
+	msgContainer   IMsgContainer
+	operatorShares map[uint32]*bls.SecretKey
```
Comment (gl-rockx) [inline dkg/frost/state.go]: Add field level description.

File: dkg/frost/instance.go (hunk 1 above)
Comment (aayush-rockx) [reply dkg/frost/instance.go]: Agree, will add this change

**Review by gl-rockx — APPROVED** [review]
